src/bindings/python/flux/cli/base.py

```diff
@@ -1046,18 +1046,17 @@ def jobspec_create(self, args):
             for keyval in args.setattr:
                 key, val = parse_jobspec_keyval("--setattr", keyval)
 
-                #  If key does not explicitly start with ".", "system."
+                #  If key does not explicitly start with ".", "attributes.", "system."
                 #   or "user.", then "system." is implied. This is a
                 #   meant to be a usability enhancement since almost all
                 #   uses of --setattr will target attributes.system.
                 #
-                if not key.startswith((".", "user.", "system.")):
+                if not key.startswith((".", "attributes.", "user.", "system.")):
                     key = "system." + key
+                elif key.startswith("."):
+                    key = "attributes" + key
 
-                #  Allow key to begin with "." which simply forces the key
-                #   to start at the top level (since .system is not applied
-                #   due to above conditional)
-                jobspec.setattr(key.lstrip("."), val)
+                jobspec.setattr(key, val)
```

Comment:
Actually it did break a few tests, which is why it's in the same commit.  For example, `--setattr=.foo` wrote to `attributes.foo` before, but now it would write to `attributes.system.foo`.

Reply:
Hmm, seems like `--setattr=.foo` should continue to have the same behavior?

Reply:
> Hmm, seems like --setattr=.foo should continue to have the same behavior?

I assume you mean w/o changes to the cli code?  This is the current code w/o my tweaks

```
                #  If key does not explicitly start with ".", "system."
                #   or "user.", then "system." is implied. This is a
                #   meant to be a usability enhancement since almost all
                #   uses of --setattr will target attributes.system.
                #
                if not key.startswith((".", "user.", "system.")):
                    key = "system." + key

                #  Allow key to begin with "." which simply forces the key
                #   to start at the top level (since .system is not applied
                #   due to above conditional)
                jobspec.setattr(key.lstrip("."), val)
```

I believe `--setattr=.foo=bar` would ultimately send `jobspec.setattr("foo", "bar)`.  Under old code that would configure `attributes.foo`, but under new code it would tweak `attributes.system.foo`, thus necessitating changes to cli handling of `--setattr`.

Reply:
Thanks, I understand now. :+1: